### PR TITLE
Change from docker to composite action to support all os workflows

### PR DESCRIPTION
**Pull request by ErebusZ**

### CODE REVIEW

**Comment by 3asm** [comment]
Has this been properly tested @ErebusZ ? this can break many users.

File: action.yml
Comment (mohsinenar) [inline action.yml:1]: This looks good but needs to be used and tested from onther repo, as if we break this version it's going to be broken for all the users that are using it now!

**Review by mohsinenar — DISMISSED** [review]

**Comment by ErebusZ** [comment]
> Has this been properly tested @ErebusZ ? this can break many users.

image

I was abble to start multiple scans on ostorlab official.

File: action.yml
Comment (ErebusZ) [reply action.yml:1]: image

image

I started a scan from this repo (https://github.com/ErebusZ/ostorlab_insecure_android_app/)

File: action.yml [hunk 1]
```diff
@@ -34,8 +34,33 @@ outputs:
...
+      shell: bash
+      env:
+        INPUT_OSTORLAB_API_KEY: ${{ inputs.ostorlab_api_key }}
+        INPUT_SCAN_TITLE: ${{ inputs.scan_title }}
+        INPUT_SCAN_PROFILE: ${{ inputs.scan_profile }}
+        INPUT_BREAK_ON_RISK_RATING: ${{ inputs.break_on_risk_rating }}
+        INPUT_MAX_WAIT_MINUTES: ${{ inputs.max_wait_minutes }}
+        INPUT_EXTRA: ${{ inputs.extra }}
+        INPUT_ASSET_TYPE: ${{ inputs.asset_type }}
+        INPUT_TARGET: ${{ inputs.target }}
```
Comment (oussamaessaji) [inline action.yml:62]: where are the credentials??

Comment (ErebusZ) [reply action.yml:62]: what credentials?

Comment (oussamaessaji) [reply action.yml:62]: ci-scan run accept creds

Comment (ErebusZ) [reply action.yml:62]: Those are passed thru `extra`

**Review by amine3 — APPROVED** [review]

**Pull request merged**